File: onebot/src/main/java/cn/evole/onebot/sdk/action/ActionList.java
package cn.evole.onebot.sdk.action;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.LinkedList;

/**
 * Description:
 * Author: cnlimiter
 * Date: 2022/9/14 17:40
 * Version: 1.0
 */
@Data
public class ActionList<T>{

    @SerializedName("status")
    public String status;

    @SerializedName("retcode")
    public int retCode;

    @SerializedName("data")
    public LinkedList<T> data;

}

File: onebot/src/main/kotlin/sdk/action/ActionList.kt
package cn.evolvefield.onebot.sdk.action

import com.google.gson.annotations.SerializedName
import lombok.Data
import java.util.LinkedList

@Data
class ActionList<T : Any> {
    @SerializedName("status")
    var status = "failed"
    @SerializedName("retcode")
    var retCode: Int = 1004
    @SerializedName("data")
    var data: LinkedList<T>? = null
}
